orientdb/src/main/scala/akka/stream/alpakka/orientdb/javadsl/OrientDBSinkSettings.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.orientdb.javadsl
+
+import akka.stream.alpakka.orientdb._
+import scaladsl.{OrientDBSinkSettings => ScalaOrientDBSinkSettings}
+
+final class OrientDBSinkSettings(val maxPartitionSize: Int = Runtime.getRuntime.availableProcessors(),
+                                 val maxPoolSize: Int = -1,
+                                 val maxRetry: Int = 1,
+                                 val retryInterval: Int = 5000,
+                                 val bufferSize: Int,
+                                 val dbUrl: String,
+                                 val username: String,
+                                 val password: String) {
+  def this(dbUrl: String, username: String, password: String) =
+    this(Runtime.getRuntime.availableProcessors(), -1, -1, 5000, 10, dbUrl, username, password)
+
+  def withMaxPartitionSize(maxPartitionSize: Int): OrientDBSinkSettings =
+    new OrientDBSinkSettings(
+      maxPartitionSize,
+      this.maxPoolSize,
+      this.maxRetry,
+      this.retryInterval,
+      this.bufferSize,
+      this.dbUrl,
+      this.username,
+      this.password
+    )
+
+  def withMaxPoolSize(maxPoolSize: Int): OrientDBSinkSettings =
+    new OrientDBSinkSettings(
+      this.maxPartitionSize,
+      maxPoolSize,
+      this.maxRetry,
+      this.retryInterval,
+      this.bufferSize,
+      this.dbUrl,
+      this.username,
+      this.password
+    )
+
+  def withRetries(maxRetry: Int, retryInterval: Int): OrientDBSinkSettings =
+    new OrientDBSinkSettings(
+      this.maxPartitionSize,
+      this.maxPoolSize,
+      maxRetry,
+      retryInterval,
+      this.bufferSize,
+      this.dbUrl,
+      this.username,
+      this.password
+    )
+
+  def withBufferSize(bufferSize: Int): OrientDBSinkSettings =
+    new OrientDBSinkSettings(
+      this.maxPartitionSize,
+      this.maxPoolSize,
+      this.maxRetry,
+      this.retryInterval,
+      bufferSize,
+      this.dbUrl,
+      this.username,
+      this.password
+    )
+
+  def withOrientDBCredentials(dbUrl: String, username: String, password: String): OrientDBSinkSettings =
+    new OrientDBSinkSettings(
+      this.maxPartitionSize,
+      this.maxPoolSize,
+      this.maxRetry,
+      this.retryInterval,
+      this.bufferSize,
+      dbUrl,
+      username,
+      password
+    )
+
+  private[javadsl] def asScala: ScalaOrientDBSinkSettings =
+    new ScalaOrientDBSinkSettings(
```

Comment:
Why did you implement settings for both Java and Scala? Could this be done in one case class containing `withXxx` methods?